staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -139,6 +139,33 @@ func NewAvailableConditionController(
 	return c
 }
 
+func setConditionAndUpdateStatus(client apiregistrationclient.APIServicesGetter, apiService *apiregistration.APIService, newCondition apiregistration.APIServiceCondition) error {
+	orig := apiService.DeepCopy()
+	apiregistration.SetAPIServiceCondition(apiService, newCondition)
+	if !reflect.DeepEqual(orig, apiService) {
```

Comment:
Just to clarify, are you checking for equality here because you only want this block to execute if a condition has changed?

Reply:
Yes. The controller (old code btw.) is level based. So we only update the status if something actually changed, but we run into the func on each sync.